src/main/java/org/logstash/dissect/search/DelimiterSearchStrategy.java

```diff
@@ -0,0 +1,5 @@
+package org.logstash.dissect.search;
+
+public interface DelimiterSearchStrategy {
```

Comment:
`DelimiterSearchStrategy` is a very long name and it is not necessary to encode the name of the pattern in the interface name. Maybe `DelimiterDetector`? I also think that the interface is quite low-level. Would your code benefit from a higher-level API like an Iterator-like interface that gives you all positions?

The next odd thing is that you operate on bytes instead of characters. As UTF-8 has multibyte characters this may be more convenient. Did you check that everything works as expected with multibyte delimiters?